tests/test_plugins/test_resonance_finder.py

```diff
@@ -0,0 +1,65 @@
+import pytest
+import numpy as np
+import pydantic
+
+import gdspy
+import tidy3d as td
+
+from numpy.random import default_rng
+
+from tidy3d.plugins import ResonanceFinder
+
+
+def try_resonance_finder(freqs, decays, amplitudes, phases, f_min, f_max):
+    freqs = np.sort(freqs)
+
+    rtol = 1e-3
+
+    t = np.linspace(0, 10000, 10000)
+    signal = np.zeros(len(t), dtype=complex)
+
+    complex_amplitudes = amplitudes * np.exp(1j * phases)
+    complex_freqs = 2 * np.pi * freqs - 1j * decays
+
+    for i in range(len(freqs)):
+        signal += complex_amplitudes[i] * np.exp(-1j * complex_freqs[i] * t)
+    resonance_finder = ResonanceFinder(freq_window=(f_min, f_max))
+    resonances = resonance_finder.run(signal, 1)
+
+    assert len(freqs) == resonances.dims["freq"]
+    for i in range(len(freqs)):
+        resonance = resonances.isel(freq=i)
+        assert np.isclose(resonance.freq, freqs[i], rtol=rtol, atol=0)
+        assert np.isclose(resonance.decay, decays[i], rtol=rtol, atol=0)
+        assert np.isclose(resonance.amplitude, amplitudes[i], rtol=rtol, atol=0)
+        assert np.isclose(
+            resonance.amplitude * np.exp(1j * resonance.phase),
+            complex_amplitudes[i],
+            rtol=rtol,
+            atol=0,
+        )
+
+
+def test_random_sinusoids():
+    """tests resonance solver on a random sum of sinusoids"""
+    rng = default_rng(12345)
```

Comment:
it's generally recommended that unit tests are deterministic (not random) so they can be easily reproduced. if something breaks. So perhaps we can set a random seed internally?

Additionally, it might be good to perform this test on a collection of different values.
To do this, you can leverage `pytest.mark.parameterize()`

```py
@pytest.mark.parameterize('rng_seed', np.arange(0, 10))
def test_random_sinusoids(rng_seed):
     np.random.seed(rng_seed)
     # ... rest of function stays the same
```